src/ui/Combobox/ComboboxValues.ts

```diff
@@ -0,0 +1,189 @@
+    this.values = this.combobox.options.createValuesFromResponse(response);
+    this.render();
+    this.updateAccessibilityAttributes();
+  }
+
+  private render() {
+    $$(this.element).show();
+
+    if (!this.hasValues()) {
+      return this.renderNoValuesFound();
+    }
+
+    this.renderValues();
+    this.addEventListeners();
+  }
+
+  private renderValues() {
+    const fragment = document.createDocumentFragment();
+    this.values.forEach((value, index) => {
+      const elementWrapper = $$('li', { id: `${this.combobox.id}value-${index}`, className: 'coveo-combobox-value' }, value.element).el;
+      value.element = elementWrapper;
+      fragment.appendChild(value.element);
+    });
+
+    this.element.appendChild(fragment);
+  }
+
+  private hasValues() {
+    return !!this.values.length;
+  }
+
+  private renderNoValuesFound() {
+    const label = this.combobox.options.noValuesFoundLabel;
+    const noValuesFoundElement = $$(
+      'li',
+      {
+        className: 'coveo-combobox-value-not-found'
+      },
+      label
+    ).el;
+
+    this.element.appendChild(noValuesFoundElement);
+    this.combobox.updateAriaLive(label);
+  }
+
+  private addEventListeners() {
+    this.values.forEach(({ element }) => {
+      $$(element).on('mouseenter', () => (this.mouseIsOverValue = true));
+      $$(element).on('mouseleave', () => (this.mouseIsOverValue = false));
+      $$(element).on('click', this.onValueClick.bind(this));
+    });
+  }
+
+  private onValueClick(e: MouseEvent) {
+    if (!this.combobox.options.selectValueOnClick) {
+      return;
+    }
+
+    const target = <HTMLElement>e.target;
+    const targetElement = !$$(target).hasClass('coveo-combobox-value') ? $$(target).parent('coveo-combobox-value') : target;
+
+    if (!targetElement) {
+      return;
+    }
+
+    const value = find(this.values, ({ element }) => element.getAttribute('id') === targetElement.getAttribute('id'));
+    value && this.combobox.options.onSelectValue(value);
+    this.combobox.clearAll();
+  }
+
+  private updateAccessibilityAttributes() {
+    const activeDescendant = this.keyboardActiveValue ? this.keyboardActiveValue.element.getAttribute('id') : '';
+
+    this.combobox.updateAccessibilityAttributes({
+      activeDescendant,
+      expanded: this.hasValues()
+    });
+  }
+
+  public clearValues() {
+    this.mouseIsOverValue = false;
+    this.resetKeyboardActiveValue();
+    $$(this.element).empty();
+    $$(this.element).hide();
+    this.values = [];
+    this.updateAccessibilityAttributes();
+  }
+
+  private setKeyboardActiveValue(value: IComboboxValue) {
+    this.keyboardActiveValue = value;
+    this.activateFocusOnValue(this.keyboardActiveValue);
+  }
+
+  private resetKeyboardActiveValue() {
+    if (!this.keyboardActiveValue) {
+      return;
+    }
+
+    this.deactivateFocusOnValue(this.keyboardActiveValue);
+    this.keyboardActiveValue = null;
+  }
+
+  private activateFocusOnValue({ element }: IComboboxValue) {
```

Comment:
why is the element in `{ }`?
I assume it creates an object with `element` but I don't get why you need to do it.

Reply:
It's just destructuring in Javascript, a quite new and pretty cool feature. Since I only need to element attribute of the IComboboxValue, I can get it straight there